app/controllers/profiles_controller.rb

```diff
@@ -13,13 +13,9 @@ def update
   end
 
   def search
-    @results = []
-    Profile.all.each do |profile|
-      if (@profile.distance_to(profile.postcode) <= params[:distance].to_f) && (profile != @profile)
-        @results << [profile, @profile.distance_to(profile.postcode).round(2)]
-      end
-    end 
-     @results.sort!{|a, b| a[1] <=> b[1]}
+    @results = Profile.near(@profile.postcode, params[:distance].to_f).to_a
```

Comment:
Just noticed you have a customer id, how about
`Profile.near(@profile.postcode, params[:distance].to_f).where("customer_id != ?" current_customer.id)` ?